lib/service/idea_service.dart

```diff
@@ -0,0 +1,193 @@
+    await ideaRef.doc(_idea.id).update({'description': description});
+    _idea = _idea.copyWith(description: description);
+    updateUpdatedAt();
+    debugPrint('updated idea ${_idea.id } description DB');
+  }
+
+  /// updates _idea udpatedAt in the database and updates _idea with that update
+  Future<void> updateUpdatedAt() async {
+    await ideaRef.doc(_idea.id).update({'updatedAt': DateTime.now().microsecondsSinceEpoch});
+    _idea = _idea.copyWith(updatedAt: DateTime.now().microsecondsSinceEpoch);
+    debugPrint('updated idea ${_idea.id } updatedAt DB');
+  }
+
+  /// updates _idea in the database and udpates _idea with that update
+  Future<void> updateVotes(String vote) async {
+    await ideaRef.doc(_idea.id).update({
+      "votes": FieldValue.arrayUnion([vote])
+    }).then((_) {
+      updateUpdatedAt();
+      debugPrint('Added vote: $vote to idea ${_idea.id}');
+    });
+
+  }
+
+  /// The idea here is to get a document ID from a Stream of multiple documents
+  /// then store it to the current _idea for use of display and updates for Idea Services.
+  ///
+  /// To get the document ID to pass to this function:
+  ///  1. StreamBuilder -> store as AsyncSnapshot<QuerySnapshot> snapshot using
+  ///    getAllIdeasFromDBStream(), getIdeasFromDBForCurrentMonthStream(), or searchIdeasByTitle()
+  ///  2. ListView -> children: snapshot.data.docs.map((document) widgets for each document)
+  ///  3. access a specific idea "onTapped" with this method using "document.id".
+  getIdeaFromDBDocument(String documentId) async{
+    DocumentSnapshot doc;
+    await ideaRef.doc(documentId).get().then((doc) {
+      if (doc.exists) {
+        print('Document data: ${doc.data()}');
+        _idea = _idea.copyWith(id: doc.id,
+            title: doc.data()["title"],
+            description: doc.data()["description"],
+            createdAt: doc.data()["createdAt"],
+            updatedAt: doc.data()["updatedAt"],
+            votes: doc.data()['categories'] as List<String>);
+      } else {
+        print('Document does not exist on the database');
+      }
+    });
+    ideaToString();
+    return doc;
+  }
+
+  ///get _idea as a stream, returns AsyncSnapshot<QuerySnapshot>
+  getCurrentIdeaFromDBAsStream() async{
+    debugPrint('getCurrentIdeaFromDBStream() performing...');
+    return ideaRef.doc(_idea.id).get().asStream();
+  }
+
+  ///get all idea documents from database,returns stream of AsyncSnapshot<QuerySnapshot>
+  getAllIdeasFromDBStream() async {
+    debugPrint('getAllIdeasFromDBStream() performing...');
+    return ideaRef.snapshots();
+  }
+
+  /// returns all ideas that were created within the last month as stream of AsyncSnapshot<QuerySnapshot>
+  getIdeasFromDBForCurrentMonthStream() async {
+    debugPrint('getIdeasFromDBForCurrentMonthStream() performing...');
+    return ideaRef.where('createdAt', isGreaterThanOrEqualTo: DateTime.now().microsecondsSinceEpoch - 2629743 );
```

Comment:
I imagine this number at the end represents a month. But if we run the function on Jan 2nd, we only want it to return results from Jan 1st onward. If we run it on Jan 13th it should return results from Jan 1st onwards. etc etc